keycloak/README.md

```diff
@@ -0,0 +1,158 @@
+    - **UDP**: If UDP protocol is used for syslog forwarding, modify the type to `udp`.
+
+    ```yaml
+      logs:
+      - type: <tcp/udp>
+        port: <PORT>
+        source: keycloak
+        service: keycloak
+    ```
+    **Note**: 
+      - `PORT`: port should be similar to port provided in **Configure syslog message forwarding from keycloak** section.
+      - It is recommended not to change the service and source values, as these parameters are integral to the pipeline's operation.
+
+3. [Restart the Agent][2].
+
+#### Configure syslog message forwarding from keycloak
+
+  1. Connect to the remote machine where Keycloak is installed.
+  2. Navigate to the directory where Keycloak is installed (typically located at `/opt/keycloak`, depending on the configuration).
+  3. Add the following options in the start command to configure Keycloak to forward logs on the datadog-agent server and execute the same on the Keycloak server terminal.
+  ```
+    --log="syslog"
+    --log-level=org.keycloak.events:debug
+    --log-syslog-endpoint=<IP Address>:<Port>
+    --log-syslog-output=json
+  ```
+
+  Optional: To use UDP instead of TCP for syslog forwarding, include the following option in the Keycloak start command:
+
+  ```
+    --log-syslog-protocol=udp
+  ```
+
+  4. After adding the above configuration option, start command would look like below:
+  ```shell
+    bin/kc.[sh|bat] start --log="syslog" --log-syslog-endpoint=<IP Address>:<Port> --log-level=org.keycloak.events:debug --log-syslog-output=json
+  ```
+  `IP ADDRESS`: ip address where your datadog-agent is running.
+  
+  `PORT`: port number to send syslog messages.
+
+  Reference: [Keycloak Syslog Configuration][7] 
+
+### Validation
+
+[Run the Agent's status subcommand][5] and look for `keycloak` under the Checks section.
+
+## Data Collected
+
+### Log 
+
+| Format     | Event Types    |
+| ---------  | -------------- |
+| JSON | user-event, admin-event |
+
+### Metrics
+
+The Keycloak integration does not include any metrics.
+
+### Events
+
+The Keycloak integration does not include any events.
+
+### Service Checks
+
+The Keycloak integration does not include any service checks.
+
+## Troubleshooting
+
+**Permission denied while port binding:**
+
+If you see a **Permission denied** error while port binding in the Agent logs:
+
+1. Binding to a port number under 1024 requires elevated permissions. Grant access to the port using the `setcap` command:
+    ```shell
+    sudo setcap CAP_NET_BIND_SERVICE=+ep /opt/datadog-agent/bin/agent/agent
+    ```
+
+2. Verify the setup is correct by running the `getcap` command:
+
+    ```shell
+    sudo getcap /opt/datadog-agent/bin/agent/agent
+    ```
+
+    With the expected output:
+
+    ```shell
+    /opt/datadog-agent/bin/agent/agent = cap_net_bind_service+ep
+    ```
+
+    **Note**: Re-run this `setcap` command every time you upgrade the Agent.
+
+3. [Restart the Agent][2].
+
+Here is how to troubleshoot some possible issues.
```

Comment:
```suggestion
```
I think we can leave this sentence out since the above heading is called Troubleshooting

Reply:
Have updated as per suggestion.